### ISSUE TITLE
Host header gets truncated on first dot

### ISSUE THREAD

**Issue by niko** [wg/wrk#124]
wrk truncates the host header to only contain the hostname up to the first dot.
```
./wrk -c 1 -d 1 -t 1 --latency  'http://api.laut.fm/station/ende'
```
```
sudo tcpdump -i eth0 dst api.laut.fm -s 1024 -A
10:13:14.108508 IP MY_IP.51860 > 92.43.104.10.http: Flags [P.], seq 0:42, ack 1, win 229, options [nop,nop,TS val 1461370 ecr 397367094], length 42
E..^..@.@......y\+h
...P...................
..Lz..W6GET /station/ende HTTP/1.1
Host: api.
```
I tried the master branch and the 4.0.0 tag. Adding the Host header manually works.

**Comment by wg** [comment]
Hi @niko, I can't reproduce this behaviour locally. Can you describe your system configuration, i.e. OS of both the client and server, network, etc?

**Comment by niko** [comment]
The server is an nginx 1.5.1 on Ubuntu 10.04. The client Arch with kernel 3.19.2.

**Comment by wg** [comment]
I ran a short test against http://api.laut.fm/station/ende and saw the correct header being passed. Could you provide a full tcpdump file? Another idea might be to run netcat on the server, and capture the request there. Perhaps this is a firewall or VPN issue?

**Comment by niko** [comment]
It's a wrk 4.0.0 problem. It works with 3.1.2.

4.0.0 truncates to the first 4 characters, not up to the first dot.

I wrote the simplest possible ruby webserver printing the request host header to stdout:
```
 % cat s.ru
run Proc.new { |env| puts env['HTTP_HOST'] ; ['200', {'Content-Type' => 'text/html'}, []] }
 % rackup s.ru
Thin web server (v1.6.3 codename Protein Powder)
Maximum connections set to 1024
Listening on localhost:9292, CTRL+C to stop
localhost:9292
127.0.0.1 - - [23/Mar/2015:13:46:21 +0100] "GET / HTTP/1.1" 200 - 0.0006
127.:9292
127.0.0.1 - - [23/Mar/2015:13:46:51 +0100] "GET / HTTP/1.1" 200 - 0.0003
[…]
loca:9292
127.0.0.1 - - [23/Mar/2015:13:47:19 +0100] "GET / HTTP/1.1" 200 - 0.0002
[…]
mysq:9292
127.0.0.1 - - [23/Mar/2015:13:47:35 +0100] "GET / HTTP/1.1" 200 - 0.0008
```
Meanwhile in another terminal:
```
% curl -i localhost:9292
% ./wrk -c 1 -d 1 -t 1 'http://127.0.0.1:9292'
% ./wrk -c 1 -d 1 -t 1 'http://localhost:9292'
% ./wrk -c 1 -d 1 -t 1 'http://mysql.dev:9292'
```
(mysql.dev pointing to localhost via /etc/hosts)

Definetely not a firewall or VPN problem. Tried the same with a simple netcat loop. Same result. Tried it from bash and zsh (in case shell expansion or something being weird): same result. Tried the ruby/rack server within a ubuntu docker container. Same result.

I'm pretty sure it's s/th with wrk.

**Comment by wg** [comment]
Thanks! You’re right, I was depending on unspecified behavior which worked as expected in Clang but not GCC. I’ve released 4.0.1 with a fix.
